src/controller/event/handler/webhook/artifact/retention.go

```diff
@@ -85,7 +85,8 @@ func (r *RetentionHandler) IsStateful() bool {
 }
 
 func (r *RetentionHandler) constructRetentionPayload(event *event.RetentionEvent) (*model.Payload, bool, int64, error) {
-	task, err := r.RetentionController().GetRetentionExecTask(event.TaskID)
+	ctx := orm.Context()
+	task, err := r.RetentionController().GetRetentionExecTask(ctx, event.TaskID)
```

Comment:
Can we assign a retention controller to the RetentionHandler directly?

In the previous version, introduce the RetentionController function to this RetentionHandler because it's difficult to initialize a retention controller. So the RetentionController introduce to get the retention controller from the core API.

With this refactor, we can get a retention controller easily. RetentionController function is not needed.